src/compiler/checker.ts

```diff
@@ -34527,7 +34529,18 @@ namespace ts {
             checkSourceElement(node.elseStatement);
         }
 
-        function checkTestingKnownTruthyCallableType(condExpr: Expression, type: Type, body: Statement | Expression | undefined) {
+        function checkTestingKnownTruthyAwaitableType(condExpr: Expression, type: Type) {

```

Comment:
This is used *almost* in every place that `checkTestingKnownTruthyCallableType` is used:

* There's one place where `checkTestingKnownTruthyCallableType` is used (`checkBinaryExpression`) but you didn't add `checkTestingKnownTruthyAwaitableType` --- it looks like it should be used there too, no?

* If it should be used there too, then it would be simpler to combine the two functions: instead of adding `checkTestingKnownTruthyAwaitableType`, just rename the existing one something like `checkTestingKnownTruthyCallableOrAwaitableType`.

* It would also simplify the code a bit.